api/apis/space_handler.go

```diff
@@ -87,8 +101,24 @@ func (h *SpaceHandler) SpaceListHandler(w http.ResponseWriter, r *http.Request)
 	orgUIDs := parseCommaSeparatedList(r.URL.Query().Get("organization_guids"))
 	names := parseCommaSeparatedList(r.URL.Query().Get("names"))
 
-	spaces, err := h.spaceRepo.FetchSpaces(ctx, orgUIDs, names)
+	spaceRepo, err := h.spaceRepoProvider.SpaceRepoForRequest(r)
+	if err != nil {
+		if authorization.IsUnauthorized(err) {
```

Comment:
Question for my own understanding:

This just checks that the user submitted an auth header right? If they have correctly submitted an `Authorization` header it will get past this and if they lack the requisite RBAC then the call on line 119 to `FetchSpaces` will fail?

If that's the case, I'm wondering if instead of an unknown error we should check to see if the request failed due to authorization reasons and return a `403` instead in that case.

Reply:
> ... and if they lack the requisite RBAC then the call on line 119 to FetchSpaces will fail?
Not really. In order to list authorised spaces efficiently we lean on the privileged client to get them all in one shot and filter out the ones the authenticated user has been bound to. We cannot rely on RBAC as namespaces are cluster-wide resources, and therefore users (and the user-scoped client) are not expected to have permissions to list them. 

TL;DR, the call on line 119 cannot fail due to authorisation issues because that line would use the privileged client that can do anything.

Reply:
Ah yeah, I see.

I guess that's more of an issue for the future endpoints then. Thanks for the clarification!